internal/services/create_auth_user.go

```diff
@@ -18,33 +18,31 @@ type CreateAuthUserService struct {
 	models           *data.Models
 	dbConnectionPool db.DBConnectionPool
 	authManager      auth.AuthManager
-	messengerClient  message.MessengerClient
 }
 
-func NewCreateUserService(models *data.Models, dbConnectionPool db.DBConnectionPool, authManager auth.AuthManager, messengerClient message.MessengerClient) *CreateAuthUserService {
+func NewCreateUserService(models *data.Models, dbConnectionPool db.DBConnectionPool, authManager auth.AuthManager) *CreateAuthUserService {
 	return &CreateAuthUserService{
 		models:           models,
 		dbConnectionPool: dbConnectionPool,
 		authManager:      authManager,
-		messengerClient:  messengerClient,
 	}
 }
 
-func (s *CreateAuthUserService) CreateUser(ctx context.Context, newUser auth.User, uiBaseURL string) (*auth.User, error) {
+func (s *CreateAuthUserService) CreateUser(ctx context.Context, newUser auth.User, uiBaseURL string) (*auth.User, *message.Message, error) {
```

Comment:
Having a CreateUser method that returns a message to be sent by another method sounds counter-intuitive. CreateUser is meant to create a user, not to prepare an invitation message (which is most of what it does, right?).

Since you're touching this part of the code, you should improve the separation of concerns, by splitting it into two methods: `CreateUser(cts, newUser, ...)` + `SendInvitationMessage(ctx, user, ...)`. WDYT?

Reply:
Looks like this was addressed